quickwit/quickwit-metastore/src/metastore/mod.rs

```diff
@@ -249,13 +267,284 @@ pub trait Metastore: Send + Sync + 'static {
-    /// to the given `delete_opstamp`.
-    async fn list_stale_splits(
+    /// Lists [`DeleteTask`] with `delete_task.opstamp` > `opstamp_start` for a given `index_id`.
+    async fn list_delete_tasks(
         &self,
         index_id: &str,
-        delete_opstamp: u64,
-        num_splits: usize,
-    ) -> MetastoreResult<Vec<Split>>;
+        opstamp_start: u64,
+    ) -> MetastoreResult<Vec<DeleteTask>>;
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+/// A query builder for listing splits within the metastore.
+pub struct ListSplitsQuery<'a> {
+    /// The index to get splits from.
+    pub index: &'a str,
+
+    /// The maximum number of splits to retrieve.
+    pub limit: Option<usize>,
+
+    /// The number of splits to skip.
+    pub offset: Option<usize>,
+
+    /// A specific split state(s) to filter by.
+    pub split_states: Vec<SplitState>,
+
+    /// A specific set of tag(s) to filter by.
+    pub tags: Option<TagFilterAst>,
+
+    /// The time range to filter by.
+    pub time_range: FilterRange<i64>,
+
+    /// The delete opstamp range to filter by.
+    pub delete_opstamp: FilterRange<u64>,
+
+    /// The update timestamp range to filter by.
+    pub update_timestamp: FilterRange<i64>,
+
+    /// The create timestamp range to filter by.
+    pub create_timestamp: FilterRange<i64>,
+}
+
+#[allow(unused_attributes)]
+impl<'a> ListSplitsQuery<'a> {
+    /// Creates a new [ListSplitsQuery] for a specific index.
+    pub fn for_index(index: &'a str) -> Self {
+        Self {
+            index,
+            limit: None,
+            offset: None,
+            split_states: vec![],
+            tags: None,
+            time_range: Default::default(),
+            delete_opstamp: Default::default(),
+            update_timestamp: Default::default(),
+            create_timestamp: Default::default(),
+        }
+    }
+
+    /// Sets the maximum number of splits to retrieve.
+    pub fn with_limit(mut self, n: usize) -> Self {
+        self.limit = Some(n);
+        self
+    }
+
+    /// Sets the number of splits to skip.
+    pub fn with_offset(mut self, n: usize) -> Self {
+        self.offset = Some(n);
+        self
+    }
+
+    /// Select splits which have the given split state.
+    pub fn with_split_state(mut self, state: SplitState) -> Self {
+        self.split_states.push(state);
+        self
+    }
+
+    /// Select splits which have the any of the following split state.
+    pub fn with_split_states(mut self, states: impl AsRef<[SplitState]>) -> Self {
+        self.split_states.extend_from_slice(states.as_ref());
+        self
+    }
+
+    /// Select splits which match the given tag filter.
+    pub fn with_tags_filter(mut self, tags: TagFilterAst) -> Self {
+        self.tags = Some(tags);
+        self
+    }
+
+    /// Set the field's lower bound to match values that are
+    /// *less than or equal to* the provided value.
+    pub fn with_time_range_lte(mut self, v: i64) -> Self {
```

Comment:
I'd rather be more explicit: `with_time_range_end_lte`, `with_time_range_end_lt`, `with_time_range_start_gt`, `with_time_range_start_gte`.